### PR TITLE
CTU OpenSource patch refurbishing

### PR DESCRIPTION
**Pull request by dkrupp**

### CODE REVIEW

File: lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp [hunk 1]
```diff
@@ -40,9 +40,17 @@
 #include "llvm/Support/Program.h"
 #include "llvm/Support/Timer.h"
 #include "llvm/Support/raw_ostream.h"
+#include <assert.h>
```
Comment (Xazax-hun) [inline lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp]: Use cassert instead.

File: lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp [hunk 2]
```diff
@@ -415,6 +424,21 @@ void AnalysisConsumer::storeTopLevelDecls(DeclGroupRef DG) {
   }
 }
 
+extern std::string getMangledName(const NamedDecl *ND,
```
Comment (dkrupp) [inline lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp]: remove extern

File: lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp [hunk 3]
```diff
@@ -415,6 +424,21 @@ void AnalysisConsumer::storeTopLevelDecls(DeclGroupRef DG) {
...
+                                  MangleContext *MangleCtx);
+
+void lockedWrite(const std::string &fileName, const std::string &content) {
```
Comment (dkrupp) [inline lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp]: remove this function

File: tools/clang-cmdline-arch-extractor/ClangCmdlineArchExtractor.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
+//===- ClangCmdlineArchExtractor.cpp ------------------------------------===//
```
Comment (dkrupp) [inline tools/clang-cmdline-arch-extractor/ClangCmdlineArchExtractor.cpp]: delete this file and implent this function in python

File: tools/clang-func-mapping/ClangFnMapGen.cpp [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    ItaniumMangleContext *ItaniumCtx =
+        ItaniumMangleContext::create(CI.getASTContext(), CI.getDiagnostics());
+    ItaniumCtx->setShouldForceMangleProto(true);
+    std::unique_ptr<ASTConsumer> PFC(
+        new MapFunctionNamesConsumer(CI.getASTContext(), ItaniumCtx));
+    return PFC;
+  }
+};
+
+int main(int argc, const char **argv) {
```
Comment (dkrupp) [inline tools/clang-func-mapping/ClangFnMapGen.cpp]: add help

File: tools/ctu-analysis/ctu-build.py [hunk 6]
```diff
@@ -133,11 +133,9 @@ def map_functions(params):
     command, sources, directory, clang_path, ctuindir = params
     args = get_command_arguments(command)
     dir_command = ['cd', directory]
-    funcmap_command = [os.path.join(clang_path, 'clang-func-mapping')]
-    funcmap_command.append('--ctu-dir')
-    funcmap_command.append(os.path.abspath(ctuindir))
+    funcmap_command = [os.path.join(clang_path, 'clang-func-mapping'),
+                       '--ctu-dir', os.path.abspath(ctuindir), '--']
```
Comment (gerazo) [inline tools/ctu-analysis/ctu-build.py]: Have you tried this out? Originally, sources are separated by '--' from the arguments. Now, '--' comes first and then sources and args together.

Comment (Xazax-hun) [inline tools/ctu-analysis/ctu-build.py]: You are right! It looks like PyCharm's automatic refactoring screwed it up. The morale, do not trust automatic refactoring tools.

**Comment by gerazo** [comment]
It seems to be working with producing same results on curl and openssl.

**Pull request closed without merging**